snowplow/iglu-client-embedded/schemas/com.metabase/embed_flow/jsonschema/1-0-3

```diff
@@ -0,0 +1,230 @@
+    },
+    "params": {
+      "type": [
+        "object",
+        "null"
+      ],
+      "description": "Parameters related to the artifact.",
+      "properties": {
+        "locked": {
+          "type": "integer",
+          "description": "Number of locked parameters in the artifact.",
+          "minimum": 0,
+          "maximum": 2147483647
+        },
+        "editable": {
+          "type": "integer",
+          "description": "Number of editable parameters in the artifact.",
+          "minimum": 0,
+          "maximum": 2147483647
+        },
+        "disabled": {
+          "type": "integer",
+          "description": "Number of disabled parameters in the artifact.",
+          "minimum": 0,
+          "maximum": 2147483647
+        }
+      }
+    },
+    "first_published_at": {
+      "type": [
+        "string",
+        "null"
+      ],
+      "format": "date-time",
+      "description": "The timestamp when the artifact was first published."
+    },
+    "language": {
+      "type": [
+        "string",
+        "null"
+      ],
+      "description": "The backend or view language of the artifact.",
+      "maxLength": 1024
+    },
+    "location": {
+      "type": [
+        "string",
+        "null"
+      ],
+      "enum": [
+        "code_overview",
+        "code_params",
+        "code_appearance"
+      ],
+      "description": "The location in the code where the event is triggered."
+    },
+    "code": {
+      "type": [
+        "string",
+        "null"
+      ],
+      "enum": [
+        "backend",
+        "view"
+      ],
+      "description": "Type of the language of the artifact where the event is triggered."
+    },
+    "appearance": {
+      "type": [
+        "object",
+        "null"
+      ],
+      "description": "The appearance settings of the artifact.",
+      "properties": {
+        "title": {
+          "type": "boolean",
+          "description": "Indicates if the title is visible."
+        },
+        "border": {
+          "type": "boolean",
+          "description": "Indicates if the border is visible."
+        },
+        "theme": {
+          "type": [
+            "string"
+          ],
+          "enum": [
+            "light",
+            "night",
+            "transparent"
+          ],
+          "description": "The theme of the artifact's appearance."
+        },
+        "font": {
+          "type": "string",
+          "enum": [
+            "instance",
+            "custom"
+          ],
+          "description": "The font type used in the artifact."
+        },
+        "hide_download_button": {
+          "type": [
+            "boolean",
+            "null"
+          ],
+          "description": "Indicates if the download button is hidden. It will be null on OSS instance since it's not supported."
+        },
+        "downloads": {
+          "description": "Indicates which download types are enabled, it will be null on OSS as they can't disable them",
+          "anyOf": [
```

Comment:
@heypoom can you link me the document for this field type? I'm not familiar with snowplow doc, and I couldn't find a doc for that. Also, @npretto should we be moving to use [Simple event](https://www.notion.so/metabase/Snowplow-integration-5da1f874beda4153b4fccfa6c1e77caa?pvs=4#59fb20748e3248ab832d62b9dc981a15) in the future?

Would it make sense to migrate to simple event in this PR?

Reply:
@WiNloSt Sure! Snowplow is based on Iglu's JSON schemas, and https://snowplow.github.io/schema-ddl/0.25.0/com/snowplowanalytics/iglu/schemaddl/jsonschema/Keyword$$AnyOf$.html indicates that you can use the `anyOf` keyword; here is an example of how Snowplow uses it: https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow.badrows/schema_violations/jsonschema/1-0-0#L24-L25

FYI, anyOf is a standard keyword for JSON schemas: https://json-schema.org/understanding-json-schema/reference/combining#anyOf